### PR TITLE
Fix codecov issue

### PR DESCRIPTION
**Pull request by tschechlovdev**
Fix for #2050
I've added a gradle task, that copys the resource to where they should be.

### CODE REVIEW

File: src/test/java/net/sf/jabref/logic/exporter/MSBibExportFormatTestFiles.java [hunk 1]
```diff
@@ -37,31 +37,42 @@
...
     }
 
+    private static Path convertPackageNameToPath(String packageName) {
+        return Paths.get(SRC_TEST_RESOURCES +packageName.replace('.','/')+"/");
+    }
+
     @Before
     public void setUp() throws Exception {
-        resourceDir = Paths.get(MSBibExportFormatTestFiles.class.getResource("").toURI());
+        resourceDir = convertPackageNameToPath( BibTeXMLExporterTestFiles.class.getPackage().getName());
```
Comment (boceckts) [inline src/test/java/net/sf/jabref/logic/exporter/MSBibExportFormatTestFiles.java]: Should be `MSBibExportFormatTestFiles.class.getPackage().getName()`, Copy paste mistake?

File: src/test/java/net/sf/jabref/logic/exporter/ModsExportFormatTestFiles.java [hunk 2]
```diff
@@ -54,13 +56,19 @@
 
     @Parameters(name = "{0}")
     public static Collection<String> fileNames() throws Exception {
-        try (Stream<Path> stream = Files.list(Paths.get(ModsExportFormatTestFiles.class.getResource("").toURI()))) {
-            //            stream.forEach(n -> System.out.println(n));
+        String packageName = MSBibExportFormatTestFiles.class.getPackage().getName();
```
Comment (boceckts) [inline src/test/java/net/sf/jabref/logic/exporter/ModsExportFormatTestFiles.java]: Same as previous, should be `ModsExportFormatTestFile.class.getPackage().getName()`.

File: src/test/java/net/sf/jabref/logic/importer/fileformat/BibTeXMLImporterTest.java [hunk 3]
```diff
@@ -32,7 +34,10 @@
      * @throws IOException
      */
     public List<Path> getTestFiles() throws Exception {
-        try (Stream<Path> stream = Files.list(Paths.get(BibTeXMLImporterTest.class.getResource("").toURI()))) {
+        String packageName = ModsImporterTestFiles.class.getPackage().getName();
```
Comment (boceckts) [inline src/test/java/net/sf/jabref/logic/importer/fileformat/BibTeXMLImporterTest.java]: Same as previous two comments, also applies to the other importer/exporters.

File: ModsExportFormatTestTotalPages.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<mods:modsCollection xmlns:mods="http://www.loc.gov/mods/v3" xmlns:ns2="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.loc.gov/standards/mods/v3/mods-3-6.xsd">
+    <mods:mods ID="test">
+        <mods:identifier type="citekey">test</mods:identifier>
+        <mods:genre>inbook</mods:genre>
```
Comment (boceckts) [inline ModsExportFormatTestTotalPages.xml]: Why is there a new test file? This is not part of this pr, only importer/exporter test classes should be changed.

File: src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java [hunk 5]
```diff
@@ -187,7 +185,7 @@ private void parseModsGroup(Map<String, String> fields, List<Object> modsGroup,
         }
 
         //The element subject can appear more than one time, that's why the keywords has to be put out of the for loop
-        putIfListIsNotEmpty(fields, keywords, FieldName.KEYWORDS, KEYWORD_SEPARATOR.toString());
+        putIfListIsNotEmpty(fields, keywords, FieldName.KEYWORDS, KEYWORD_SEPARATOR);
```
Comment (tschechlovdev) [inline src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java]: the keywords were changed from ", " to a char ','. Thats why the test won't work. This wasn't detected because the test weren't running.

File: src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java [hunk 6]
```diff
@@ -37,6 +38,8 @@
 @RunWith(Parameterized.class)
 public class BibTeXMLExporterTestFiles {
 
+    public static final String SRC_TEST_RESOURCES = "src/test/resources/";
```
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java]: done.

File: src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java [hunk 7]
```diff
@@ -45,23 +48,32 @@
...
+        Path path = convertPackageNameToPath(packageName);
+
+        try (Stream<Path> stream = Files.list(path)) {
             return stream.map(n -> n.getFileName().toString()).filter(n -> n.endsWith(".bib"))
                     .filter(n -> n.startsWith("BibTeXML")).collect(Collectors.toList());
         }
     }
 
+    private static Path convertPackageNameToPath(String packageName) {
+        return Paths.get(SRC_TEST_RESOURCES +packageName.replace('.','/')+"/");
```
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java]: Done.

File: src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java [hunk 8]
```diff
@@ -45,23 +48,32 @@
...
+    }
```
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java]: Done.

File: src/test/java/net/sf/jabref/logic/exporter/MSBibExportFormatTestFiles.java [hunk 9]
```diff
@@ -37,31 +37,42 @@
...
+    public static final String SRC_TEST_RESOURCES = "src/test/resources/";
+
+    private Path importFile ;
     public BibDatabaseContext databaseContext;
     public Charset charset;
     public File tempFile;
     public MSBibExportFormat msBibExportFormat;
-    public BibtexImporter testImporter;
 
+    public BibtexImporter testImporter;
```
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/exporter/MSBibExportFormatTestFiles.java]: Moved it back.

File: src/test/java/net/sf/jabref/logic/exporter/MSBibExportFormatTestFiles.java (hunk 1 above)
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/exporter/MSBibExportFormatTestFiles.java]: Yes, I've changed it.

File: src/test/java/net/sf/jabref/logic/exporter/ModsExportFormatTestFiles.java (hunk 2 above)
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/exporter/ModsExportFormatTestFiles.java]: Yes, done.

File: src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java (hunk 8 above)
Comment (Siedlerchr) [inline src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java]: I am not happy with that as it heavily relies on hard coded things. Normally the class path and resource dir should be resolved correctly. I bet there is one thing different with gradle and the resource dir vs in eclipse

File: src/test/java/net/sf/jabref/logic/importer/fileformat/BibTeXMLImporterTest.java (hunk 3 above)
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/importer/fileformat/BibTeXMLImporterTest.java]: Done.

File: ModsExportFormatTestTotalPages.xml (hunk 4 above)
Comment (tschechlovdev) [inline ModsExportFormatTestTotalPages.xml]: Done. Removed it.

File: src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java (hunk 8 above)
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/exporter/BibTeXMLExporterTestFiles.java]: @Siedlerchr yeah you're right. This problem seems to be a very well known [intelliJ problem] (http://stackoverflow.com/questions/27046642/access-file-in-junit-test-in-gradle-environment) that occurs here as well. I've added a gradle task that copys the testfiles to where they should be.

**Comment by stefan-kolb** [comment]
@tobiasdiez One for you to approve?

File: src/test/java/net/sf/jabref/logic/exporter/ModsExportFormatTestFiles.java [hunk 10]
```diff
@@ -90,13 +90,10 @@ public final void testPerformExport() throws Exception {
...
     public final void testExportAsModsAndThenImportAsMods() throws Exception {
-        String xmlFileName = filename.replace(".bib", ".xml");
-        Path xmlFile = Paths.get(ModsExportFormatTestFiles.class.getResource(xmlFileName).toURI());
         List<BibEntry> entries = bibtexImporter.importDatabase(importFile, charset).getDatabase().getEntries();
-        Path modsFile = importFile.resolve(xmlFile);
 
-        modsExportFormat.performExport(databaseContext, modsFile, charset, entries);
-        BibEntryAssert.assertEquals(entries, modsFile, modsImporter);
+        modsExportFormat.performExport(databaseContext, tempFile.getPath(), charset, entries);
+        BibEntryAssert.assertEquals(entries, Paths.get(tempFile.getPath()), modsImporter);
```
Comment (Siedlerchr) [inline src/test/java/net/sf/jabref/logic/exporter/ModsExportFormatTestFiles.java]: Why to you wrap in Paths.get() Isn't tmpFile.getPath already giving back a Path object?

File: src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java [hunk 11]
```diff
@@ -218,7 +216,7 @@ private void parseIdentifier(Map<String, String> fields, IdentifierDefinition id
     }
 
     private void parseTopic(Map<String, String> fields, List<JAXBElement<?>> topicOrGeographicOrTemporal,
-            List<String> keywords) {
+                            List<String> keywords) {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java]: I'm not sure about our conventions regarding code indention. @koppor is this ok?

**Comment by tobiasdiez** [comment]
So the problem is that the test resources are not copied on build and thus the tests don't found the test files? The gradle job does work (+2 % code coverage). If I understand the solution correctly, then a normal build in intellij or eclipse still don't copy the resource files and thus locally the importer tests are still not performed. Is this right? I have no idea about our build script, but is there no other way to always copy the resources (the language files are automatically copied, right?).

**Comment by Siedlerchr** [comment]
@tobiasdiez  If you run the test locally in Eclipse it executed all parameterized tests. However, only when executed via intellij or gradle it did not locate the correct directory. Thus no tests files were found.
See here for details: https://github.com/JabRef/jabref/issues/2050

File: src/test/java/net/sf/jabref/logic/exporter/ModsExportFormatTestFiles.java (hunk 10 above)
Comment (tschechlovdev) [inline src/test/java/net/sf/jabref/logic/exporter/ModsExportFormatTestFiles.java]: No, it's returning a string.

**Comment by Siedlerchr** [comment]
Okay, I would say I merge it in!
